### PR TITLE
dnscrypt-proxy: update to version 1.9.5 and add test

### PR DESCRIPTION
**Pull request by AndWac**
I have postponed 64 bit because of too many problems for now.
I haven't checked whether the service file also has to be changed.

### CODE REVIEW

File: components/network/dnscrypt-proxy/dnscrypt-proxy.p5m [hunk 1]
```diff
@@ -28,9 +30,26 @@ user username=dnscrypt ftpuser=false gcos-field="DNSCrypt Reserved UID" group=dn
 
 file files/dnscrypt-proxy.xml path=lib/svc/manifest/network/dns/dnscrypt-proxy.xml
 
+file path=etc/dnscrypt-proxy.conf
```
Comment (pyhalov) [inline components/network/dnscrypt-proxy/dnscrypt-proxy.p5m]: preserve=true?

Comment (AndWac) [reply components/network/dnscrypt-proxy/dnscrypt-proxy.p5m]: added

**Comment by AndWac** [comment]
How shall we proceed?
I have been able to successfully publish the 64 bit version after some help from alanc.
Shall I update this PR to 64 bit?
We would loose some of the plugin example libs, though.

**Comment by xen0l** [comment]
Move to 64bit.

**Comment by AndWac** [comment]
Ok, I updated this PR to 64 bit.

File: components/network/dnscrypt-proxy/Makefile [hunk 2]
```diff
@@ -11,17 +11,19 @@
...
 # Copyright 2016 Alexander Pyhalov
+# Copyright 2017 Andreas Wacknitz
 #
 
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= dnscrypt-proxy
-COMPONENT_VERSION= 1.6.1
+COMPONENT_VERSION= 1.9.5
+COMPONENT_REVISION=1
```
Comment (pyhalov) [inline components/network/dnscrypt-proxy/Makefile]: Please, remove COMPONENT_REVISION

File: components/network/dnscrypt-proxy/dnscrypt-proxy.p5m [hunk 3]
```diff
@@ -28,9 +29,24 @@ user username=dnscrypt ftpuser=false gcos-field="DNSCrypt Reserved UID" group=dn
 
 file files/dnscrypt-proxy.xml path=lib/svc/manifest/network/dns/dnscrypt-proxy.xml
 
-file path=usr/bin/hostip
-file path=usr/sbin/dnscrypt-proxy
+file path=etc/dnscrypt-proxy.conf preserve=yes
+file path=usr/bin/$(MACH64)/hostip
+link path=usr/bin/hostip target=$(MACH64)/hostip
```
Comment (pyhalov) [inline components/network/dnscrypt-proxy/dnscrypt-proxy.p5m]: Don't do this. Just set CONFIGURE_BINDIR.64=$(CONFIGURE_PREFIX)/bin

File: components/network/dnscrypt-proxy/dnscrypt-proxy.p5m [hunk 4]
```diff
@@ -28,9 +29,24 @@ user username=dnscrypt ftpuser=false gcos-field="DNSCrypt Reserved UID" group=dn
 
 file files/dnscrypt-proxy.xml path=lib/svc/manifest/network/dns/dnscrypt-proxy.xml
 
-file path=usr/bin/hostip
-file path=usr/sbin/dnscrypt-proxy
+file path=etc/dnscrypt-proxy.conf preserve=yes
```
Comment (pyhalov) [inline components/network/dnscrypt-proxy/dnscrypt-proxy.p5m]: preserve=true

**Comment by AndWac** [comment]
done

**Pull request merged**